receiver/prometheusreceiver/config_test.go

```diff
@@ -119,8 +121,8 @@ func TestLoadConfigFailsOnUnknownPrometheusSection(t *testing.T) {
 	require.Error(t, component.UnmarshalConfig(sub, cfg))
 }
 
-// Renaming is not allowed
-func TestLoadConfigFailsOnRenameDisallowed(t *testing.T) {
+// Renaming emits a warning
+func TestConfigWarningsOnRenameDisallowed(t *testing.T) {
```

Comment:
Would it make sense to add a test for this that goes through the main `createMetricsReceiver` to make sure we're logging once for each rename, and still hitting the method properly? My concern is just making sure it works as a whole, not just at the direct call/unit test level.